include/utils/tree_node.h

```diff
@@ -209,6 +218,17 @@ class TreeNode
   const unsigned int tgt_bin_size;
 
   /**
+   * The current refinement level of this TreeNode.
+   */
+  unsigned int refinement_level;
+
+  /**
+   * This specifies the refinement level beyond which we will
+   * scale up the target bin size in child TreeNodes.
+   */
```

Comment:
I guess you mean to add something like "We set the default to be 10 levels." at the end of this comment?